testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/testcontainers/PortBinder.java

```diff
@@ -33,12 +32,17 @@ public PortBinder(EnvironmentOptions environmentOptions)
 
     public void exposePort(DockerContainer container, int port)
     {
+        container.addExposedPort(port); // Still export port, at a random free number, as certain startup checks require this.
         if (bindPorts) {
-            ContainerUtil.exposePort(container, port);
-        }
-        else {
-            // Still export port, at a random free number, as certain startup checks require this.
-            container.addExposedPort(port);
+            container.withFixedExposedPort(port, port);
         }
     }
+
+    // This method exposed port unconditionally on the host machine.
+    // Usage should be limited to exposing debugging ports.
```

Comment:
exposed -> exposes

Capture _the why_ in the comment -- you know this, i think i know this, but i will need to ask you in a week from now